test/integration/clustercidr/ipam_test.go

```diff
@@ -138,6 +139,456 @@ func TestIPAMMultiCIDRRangeAllocatorType(t *testing.T) {
+		if _, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Create(context.TODO(), clusterCIDR, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		// Create a ClusterCIDR which has fewer matching labels than the previous ClusterCIDR.
+		clusterCIDR2 := makeClusterCIDR("few-label-match-cc-del", "10.1.0.0/23", "fd12:30:100::/119", 8, nodeSelector(map[string][]string{"ipv4": {"true"}}))
+		if _, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Create(context.TODO(), clusterCIDR2, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		clusterCIDRs := []*networkingv1alpha1.ClusterCIDR{clusterCIDR, clusterCIDR2}
+
+		for _, cc := range clusterCIDRs {
+			// Wait for the ClusterCIDR to be created.
+			if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
+				createdCC, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), cc.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				return createdCC != nil, nil
+			}); err != nil {
+				t.Fatalf("failed while waiting for ClusterCIDR %q to be created: %v", cc.Name, err)
+			}
+		}
+
+		// Sleep for one second to make sure the controller processes the newly created ClusterCIDR.
+		time.Sleep(1 * time.Second)
+
+		// Create a node, which gets pod CIDR from the clusterCIDR created above.
+		node := makeNode("dualstack-node", map[string]string{"ipv4": "true", "ipv6": "true"})
+		expectedPodCIDRs := []string{"192.168.0.0/24", "fd00:30:100::/120"}
+		if _, err := clientSet.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+		if gotPodCIDRs, err := nodePodCIDRs(clientSet, node.Name); err != nil {
+			t.Fatal(err)
+		} else if !reflect.DeepEqual(expectedPodCIDRs, gotPodCIDRs) {
+			t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", expectedPodCIDRs, gotPodCIDRs)
+		}
+
+		// Delete the ClusterCIDR
+		if err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Delete(context.TODO(), clusterCIDR.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		//Wait and make sure that the ClusterCIDR is not deleted, as there is a node associated with it.
+		if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
+			cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), clusterCIDR.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			return cc != nil, nil
+		}); err != nil {
+			t.Fatalf("failed while waiting for ClusterCIDR %q to be deleted: %v", clusterCIDR.Name, err)
+		}
+
+		//Delete the node.
+		if err := clientSet.CoreV1().Nodes().Delete(context.TODO(), node.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		// Create a node, which should get Pod CIDRs from the ClusterCIDR with fewer matching label Count,
+		// as the best match ClusterCIDR is marked as terminating.
```

Comment:
but if you have deleted the node in the previous step, the cluster cidr will be deleted, at least based on previous test, and this will race. 
It seems that you need to omit the previous step of deleting the node

Reply:
Good catch, copy pasted and then forgot